src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/ConstructionTapeHelper.java

```diff
@@ -0,0 +1,268 @@
+                    world.setBlockState(row2, ConstructionTape.withProperty(FACING, EnumFacing.NORTH));
+                }
+            }
+            if (z1 < z3)
+            {
+                for (int i = z1+1; i < z3; i++)
+                {
+                    newY = checkIfPlaceable(x1,y,i,world);
+                    final BlockPos row3 = new BlockPos(x1, newY, i);
+                    world.setBlockState(row3, ConstructionTape.withProperty(FACING, EnumFacing.EAST));
+                    newY = checkIfPlaceable(x3,y,i,world);
+                    final BlockPos row4 = new BlockPos(x3, newY, i);
+                    world.setBlockState(row4, ConstructionTape.withProperty(FACING, EnumFacing.WEST));
+                }
+            }
+            else
+            {
+                for (int i = z3+1; i < z1; i++)
+                {
+                    newY = checkIfPlaceable(x1,y,i,world);
+                    final BlockPos row3 = new BlockPos(x1, newY, i);
+                    world.setBlockState(row3, ConstructionTape.withProperty(FACING, EnumFacing.EAST));
+                    newY = checkIfPlaceable(x3,y,i,world);
+                    final BlockPos row4 = new BlockPos(x3, newY, i);
+                    world.setBlockState(row4, ConstructionTape.withProperty(FACING, EnumFacing.WEST));
+                }
+            }
+            newY = checkIfPlaceable(x1,y,z1,world);
+            final BlockPos corner1 = new BlockPos(x1, newY, z1);
+            newY = checkIfPlaceable(x1,y,z3,world);
+            final BlockPos corner2 = new BlockPos(x1, newY, z3);
+            newY = checkIfPlaceable(x3,y,z1,world);
+            final BlockPos corner3 = new BlockPos(x3, newY, z1);
+            newY = checkIfPlaceable(x3,y,z3,world);
+            final BlockPos corner4 = new BlockPos(x3, newY, z3);
+            world.setBlockState(corner1, ConstructionTapeCorner.withProperty(FACING, EnumFacing.SOUTH));
+            world.setBlockState(corner2, ConstructionTapeCorner.withProperty(FACING, EnumFacing.EAST));
+            world.setBlockState(corner3, ConstructionTapeCorner.withProperty(FACING, EnumFacing.WEST));
+            world.setBlockState(corner4, ConstructionTapeCorner.withProperty(FACING, EnumFacing.NORTH));
+        }
+
+    /**
+     ** Proxy to remove the tape also with the building only.
+     * @param building the building.
+     * @param world the world.
+     */
+    public static void removeConstructionTape(@NotNull AbstractBuilding building, @NotNull World world)
+    {
+        removeConstructionTape(new WorkOrderBuild(building, 1), world);
+    }
+
+    public static void removeConstructionTape(@NotNull WorkOrderBuild workOrder,@NotNull World world)
+    {
+        final StructureWrapper wrapper = new StructureWrapper(world, workOrder.getStructureName());
+        final BlockPos pos = workOrder.getBuildingLocation();
+        int tempRotation = 0;
+        final IBlockState Air = Blocks.AIR.getDefaultState();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "Air" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1481)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "Air". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00117)